File: packages/rslint-test-tools/tests/eslint-plugin-jest/rules/no-mocks-import.test.ts
import { RuleTester } from '../rule-tester';

const ruleTester = new RuleTester();

ruleTester.run('no-mocks-import', {} as never, {
  valid: [
    { code: 'import something from "something"' },
    { code: 'require("somethingElse")' },
    { code: 'require("./__mocks__.js")' },
    { code: 'require("./__mocks__x")' },
    { code: 'require("./__mocks__x/x")' },
    { code: 'require("./x__mocks__")' },
    { code: 'require("./x__mocks__/x")' },
    { code: 'require()' },
    { code: 'var path = "./__mocks__.js"; require(path)' },
    { code: 'entirelyDifferent(fn)' },
  ],
  invalid: [
    {
      code: 'require("./__mocks__")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'require("./__mocks__/")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'require("./__mocks__/index")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'require("__mocks__")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'require("__mocks__/")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'require("__mocks__/index")',
      errors: [{ line: 1, column: 9, messageId: 'noManualImport' }],
    },
    {
      code: 'import thing from "./__mocks__/index"',
      errors: [{ line: 1, column: 1, messageId: 'noManualImport' }],
    },
  ],
});

File: internal/plugins/jest/rules/no_mocks_import/no_mocks_import_test.go
package no_mocks_import_test

import (
	"testing"

	"github.com/web-infra-dev/rslint/internal/plugins/jest/fixtures"
	"github.com/web-infra-dev/rslint/internal/plugins/jest/rules/no_mocks_import"
	"github.com/web-infra-dev/rslint/internal/rule_tester"
)

func TestNoMocksImportRule(t *testing.T) {
	rule_tester.RunRuleTester(
		fixtures.GetRootDir(),
		"tsconfig.json",
		t,
		&no_mocks_import.NoMocksImportRule,
		[]rule_tester.ValidTestCase{
			{Code: `import something from "something"`},
			{Code: `require("somethingElse")`},
			{Code: `require("./__mocks__.js")`},
			{Code: `require("./__mocks__x")`},
			{Code: `require("./__mocks__x/x")`},
			{Code: `require("./x__mocks__")`},
			{Code: `require("./x__mocks__/x")`},
			{Code: `require()`},
			{Code: `var path = "./__mocks__.js"; require(path)`},
			{Code: `entirelyDifferent(fn)`},
		},
		[]rule_tester.InvalidTestCase{
			{
				Code: `require("./__mocks__")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `require("./__mocks__/")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `require("./__mocks__/index")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `require("__mocks__")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `require("__mocks__/")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `require("__mocks__/index")`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 9},
				},
			},
			{
				Code: `import thing from "./__mocks__/index"`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "noManualImport", Line: 1, Column: 1},
				},
			},
		},
	)
}
